### ISSUE TITLE
Memory Usage

### ISSUE THREAD

**Issue by Cmaury** [larspars/word-rnn#2]
You mention in the about article that you are running this on a GTX980. I'm running this on a GTX970 with 4gbs of ram, and am running out of memory using the same Glove vectors and a data set of 104mb on the default word-level options.

Are you running on a 980ti with 12gbs of memory? Is there something else at play?

Thanks, and huge thanks for making this code available.

**Comment by larspars** [comment]
I should probably said something about this. I did run this on a GTX980 with 4gbs of ram. What I did  was to _drastically_ cut down on the vocabulary size. There is a command line option for that. I kept the vocabulary just below < 2000 words - it still works OK for generating text at least. Another big factor is the number of timesteps the network is unrolled. I used 35 timesteps - I suspect this is actually way more than what is necessary.

**Comment by Cmaury** [comment]
Ah, thanks. This helps to clarify things; though I do not see an option to limit the vocab_size. Looking through train.lua, it looks like vocab_size is just set to the length of the vocab file.

edit: I figured it out. You were referring to the threshold option, which you would run against to see how large the vocab would be. The threshold will vary data set to data set depending on how large it is and how often words occur.

Separate from this, I've lowered the number of time steps. and enabled using the Glove embedding, which was not the default.
